packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -67,49 +71,67 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const [ homeId, setHomeId ] = useState();
+	useEffect( () => {
+		const effect = async () => {
+			try {
+				const { success, data } = await fetch(
+					addQueryArgs( '/', { '_wp-find-template': true } )
+				).then( ( res ) => res.json() );
+				if ( success ) {
+					let newHomeId = data.ID;
+					if ( newHomeId === null ) {
+						const { getEntityRecords } = resolveSelect( 'core' );
+						newHomeId = (
+							await getEntityRecords( 'postType', 'wp_template', {
+								resolved: true,
+								slug: data.post_name,
+							} )
+						 )[ 0 ].id;
+					}
+					setHomeId( newHomeId );
+				} else {
+					throw new Error();
+				}
+			} catch ( err ) {
+				setHomeId( null );
+			}
+		};
+		effect();
+	}, [] );
+
 	const { currentTheme, templates, templateParts } = useSelect(
-		( select ) => {
-			const { getCurrentTheme, getEntityRecord } = select( 'core' );
+		( _select ) => {
+			const { getCurrentTheme, getEntityRecords } = _select( 'core' );
+
 			return {
 				currentTheme: getCurrentTheme(),
-				templates: ids.map( ( id ) => {
-					const template = getEntityRecord(
-						'postType',
-						'wp_template',
-						id
-					);
-					return {
-						label: template ? (
-							<TemplateLabel
-								template={ template }
-								homeId={ homeId }
-							/>
-						) : (
-							__( 'Loading…' )
-						),
-						value: id,
-						slug: template ? template.slug : __( 'Loading…' ),
-					};
-				} ),
-				templateParts: templatePartIds.map( ( id ) => {
-					const template = getEntityRecord(
-						'postType',
-						'wp_template_part',
-						id
-					);
-					return {
-						label: template ? (
-							<TemplateLabel template={ template } />
-						) : (
-							__( 'Loading…' )
-						),
-						value: id,
-						slug: template ? template.slug : __( 'Loading…' ),
-					};
-				} ),
+				templates: getEntityRecords( 'postType', 'wp_template', {
+					resolved: true,
+				} )?.map( ( template ) => ( {
+					label: (
+						<TemplateLabel
+							template={ template }
+							homeId={ homeId }
+						/>
+					),
+					value: template.id,
+					slug: template.slug,
+				} ) ),
+				templateParts: getEntityRecords(
+					'postType',
+					'wp_template_part',
+					{
+						resolved: true,
+						theme: getCurrentTheme()?.stylesheet,
```

Comment:
`getCurrentTheme` is called twice. We also can't call `getEntityRecords` until it is available. Otherwise, we might make an intermediary request without a theme value.

Reply:
Should be fixed by 6e26ef7